### PR TITLE
[CARBONDATA-1445] Fix update fail when carbon.update.persist.enable'='false'

### PR DESCRIPTION
**Pull request by ravipesala**
The UDF for getting segementid while loading the data is not handled so when it needs to reexecute the rdd when persist enable is false it is not getting tupleId from carbon

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/580/

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/582/

**Comment by jackylk** [comment]
LGTM

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala [hunk 1]
```diff
@@ -448,6 +448,40 @@ class UpdateCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
...
+      .format("carbondata")
+      .option("tableName", "study_carbondata")
+      .option("compress", "true")  // just valid when tempCSV is true
+      .option("tempCSV", "false")
+      .option("single_pass", "true")
+      .option("sort_scope", "LOCAL_SORT")
+      .mode(SaveMode.Append)
+      .save()
+    sql("""
+      UPDATE default.study_carbondata a
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala]: Please add assert to check the updated value

**Comment by zzcclp** [comment]
LGTM, it works correctly now.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala (hunk 1 above)
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala]: Added

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/616/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/619/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/621/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/626/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/642/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/643/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/649/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/655/

**Comment by zzcclp** [comment]
@ravipesala @chenliang613 will this pr be merged into branch-1.2?

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/702/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/703/

**Comment by zzcclp** [comment]
retest this please

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/97/

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder1/3/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/729/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/125/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/126/

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder1/23/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/141/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/15/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/157/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/785/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/166/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/41/

**Comment by zzcclp** [comment]
what's problem with this pr, it always builds failed with Spark 2.1.0?

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/794/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/48/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/172/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/802/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/60/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/183/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/814/

**Pull request closed without merging**